Don't create the repository directory

joshuaspence opened the pull request:
There's no need to create the repository directory (`/var/repo`) in Puppet, because the repository-pull daemon will create the directory itself:

```
try {
  $dirname = dirname($local_path);
  if (!Filesystem::pathExists($dirname)) {
    Filesystem::createDirectory($dirname, 0755, $recursive = true);
  }

  if (!Filesystem::pathExists($local_path)) {
    $this->logPull(
      pht(
        'Creating a new working copy for repository "%s".',
        $repository->getDisplayName()));
    if ($is_git) {
      $this->executeGitCreate();
    } else if ($is_hg) {
      $this->executeMercurialCreate();
    } else {
      $this->executeSubversionCreate();
    }
  }

  // ...
}
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/joshuaspence/puppet-phabricator/8)
<!-- Reviewable:end -->

jarro2783 commented:
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[manifests/init.pp, line 71 at r1](https://reviewable.io:443/reviews/joshuaspence/puppet-phabricator/8#-KoLQ0KhazDGXNrsLo6Y:-KoLQ0KhazDGXNrsLo6Z:be66ojx) ([raw file](https://github.com/joshuaspence/puppet-phabricator/blob/092b015ce3009e9be4c8601958201765e9f7927d/manifests/init.pp#L71)):*
> ```Puppet
>       'phd.pid-directory' => $pid_dir,
>       'phd.user' => $daemon_user,
>       'repository.default-local-path' => $repo_dir,
> ```

How do you set a different repository directory now?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/joshuaspence/puppet-phabricator/8)*
<!-- Sent from Reviewable.io -->

joshuaspence commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[manifests/init.pp, line 71 at r1](https://reviewable.io:443/reviews/joshuaspence/puppet-phabricator/8#-KoLQ0KhazDGXNrsLo6Y:-KoLR4OPfDsu482JnyQu:b848eth) ([raw file](https://github.com/joshuaspence/puppet-phabricator/blob/092b015ce3009e9be4c8601958201765e9f7927d/manifests/init.pp#L71)):*
<details><summary><i>Previously, jarro2783 wrote…</i></summary><blockquote>

How do you set a different repository directory now?
</blockquote></details>

With `$phabricator::config_hash`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/joshuaspence/puppet-phabricator/8)*
<!-- Sent from Reviewable.io -->

jarro2783 commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[manifests/init.pp, line 71 at r1](https://reviewable.io:443/reviews/joshuaspence/puppet-phabricator/8#-KoLQ0KhazDGXNrsLo6Y:-KoLepHTV-jHAWCFPkQm:b-3g1da2) ([raw file](https://github.com/joshuaspence/puppet-phabricator/blob/092b015ce3009e9be4c8601958201765e9f7927d/manifests/init.pp#L71)):*
<details><summary><i>Previously, joshuaspence (Joshua Spence) wrote…</i></summary><blockquote>

With `$phabricator::config_hash`
</blockquote></details>

Yeah right ok.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/joshuaspence/puppet-phabricator/8)*
<!-- Sent from Reviewable.io -->

joshuaspence committed:
Don't create the repository directory

There's no need to create the repository directory (`/var/repo`) in Puppet, because the repository-pull daemon will create the directory itself:

```
try {
  $dirname = dirname($local_path);
  if (!Filesystem::pathExists($dirname)) {
    Filesystem::createDirectory($dirname, 0755, $recursive = true);
  }

  if (!Filesystem::pathExists($local_path)) {
    $this->logPull(
      pht(
        'Creating a new working copy for repository "%s".',
        $repository->getDisplayName()));
    if ($is_git) {
      $this->executeGitCreate();
    } else if ($is_hg) {
      $this->executeMercurialCreate();
    } else {
      $this->executeSubversionCreate();
    }
  }

  // ...
}
```
